src/kvirc/sparser/KviIrcServerParser_literalHandlers.cpp

```diff
@@ -954,9 +954,13 @@ void KviIrcServerParser::parseLiteralPrivmsg(KviIrcMessage * msg)
 		//			}
 		//		}
 
+		// "znc.in/self-message" capability: Handle a replayed message from ourselves to someone else.
+		bool bSelfMessage = IS_ME(msg, szNick);
+		QString szWindow = bSelfMessage ? szTarget : szNick;
```

Comment:
Not for me. I'm guessing that in your case, the self-message PRIVMSG is addressed to the full nick!user@host, instead of just the nick?